test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -1353,6 +1411,436 @@ TEST_F(OpenTelemetryPluginNPCMetricsTest,
+                                                      kOptionalLabelValues),
+                                         GaugeResultEq(kInitialValue))))));
+  auto initial_timestamp = opentelemetry::nostd::get<
+                               opentelemetry::sdk::metrics::LastValuePointData>(
+                               data[kMetricName][0].point_data)
+                               .sample_ts_;
+  stats_plugins.SetGauge(handle, kSkippedValue, kLabelValues,
+                         kOptionalLabelValues);
+  stats_plugins.SetGauge(handle, kFinalValue, kLabelValues,
+                         kOptionalLabelValues);
+  data = ReadCurrentMetricsData(
+      [&](const absl::flat_hash_map<
+          std::string,
+          std::vector<opentelemetry::sdk::metrics::PointDataAttributes>>&
+              data) { return !data.contains(kMetricName); });
+  EXPECT_THAT(data,
+              ::testing::ElementsAre(::testing::Pair(
+                  kMetricName,
+                  ::testing::ElementsAre(::testing::AllOf(
+                      AttributesEq(kLabelKeys, kLabelValues, kOptionalLabelKeys,
+                                   kOptionalLabelValues),
+                      GaugeResultEq(kFinalValue),
+                      GaugeResultLaterThan(initial_timestamp))))));
+}
+
+TEST_F(OpenTelemetryPluginNPCMetricsTest, SetDoubleGauge) {
+  constexpr absl::string_view kMetricName = "double_gauge";
+  constexpr double kInitialValue = 3.14;
+  constexpr double kSkippedValue = 5.28;
+  constexpr double kFinalValue = 7.42;
+  constexpr std::array<absl::string_view, 2> kLabelKeys = {"label_key_1",
+                                                           "label_key_2"};
+  constexpr std::array<absl::string_view, 2> kOptionalLabelKeys = {
+      "optional_label_key_1", "optional_label_key_2"};
+  constexpr std::array<absl::string_view, 2> kLabelValues = {"label_value_1",
+                                                             "label_value_2"};
+  constexpr std::array<absl::string_view, 2> kOptionalLabelValues = {
+      "optional_label_value_1", "optional_label_value_2"};
+  auto handle = grpc_core::GlobalInstrumentsRegistry::RegisterDoubleGauge(
+      kMetricName, "A simple double gauge.", "unit", kLabelKeys,
+      kOptionalLabelKeys,
+      /*enable_by_default=*/true);
+  Init(std::move(Options()
+                     .set_metric_names({kMetricName})
+                     .set_target_selector([](absl::string_view target) {
+                       return absl::StartsWith(target, "dns:///");
+                     })
+                     .add_optional_label(kOptionalLabelKeys[0])
+                     .add_optional_label(kOptionalLabelKeys[1])));
+  auto stats_plugins =
+      grpc_core::GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
+          grpc_core::StatsPlugin::ChannelScope("dns:///localhost:8080", ""));
+  stats_plugins.SetGauge(handle, kInitialValue, kLabelValues,
```

Comment:
These tests should all include setting the value for different sets of labels.

Reply:
Done.